neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/embedded/dialect/impl/EmbeddedNeo4jBackendQueryResultIterator.java

```diff
@@ -64,6 +68,13 @@ private TupleSnapshot createSnapshot(Map<String, Object> next) {
 
 	private TupleSnapshot mapSnapshot(Map<String, Object> next) {
 		TupleSnapshot snapshot;
+		if ( this.columns != null ) {
+			Map<String, Object> sortedColumns = new LinkedHashMap<>();
+			for ( String column : this.columns ) {
+				sortedColumns.put( column, next.get( column ) );
+			}
+			next = sortedColumns;
+		}
```

Comment:
Could you create a test for this? Just in case somebody in the future decide to change it? ALso, are there any test failures in the build if this is not committed? Thanks

Reply:
Yes, I've created a test which will fail if this code is removed. Also PositionalParametersStoredProcedureCallTest.testResultSetStaticCallRaw will fail, because in this test we have no EntityMetadataInformation or no other source of information about proper column order

Reply:
Thanks